Explorer/Assets/DCL/SDKComponents/Tween/Systems/TweenLoaderSystem.cs

```diff
@@ -30,14 +28,19 @@ private void LoadTween(in Entity entity, ref PBTween pbTween)
         {
             if (pbTween.ModeCase == PBTween.ModeOneofCase.None) return;
 
-            SDKTweenComponent sdkTweenComponent = new SDKTweenComponent
+            if (pbTween.ModeCase == PBTween.ModeOneofCase.TextureMove)
             {
-                IsDirty = true
+                var sdkTweenTextureComponent = new SDKTweenTextureComponent(pbTween.TextureMove.MovementType);
+
+                World.Add(entity, sdkTweenTextureComponent);
```

Comment:
A very small optimization that could be done here is declaring this above this if
```
  var sdkTweenComponent = new SDKTweenComponent
            {
                IsDirty = true,
            };
```
And inside the `if `do                 
```
World.Add(entity, sdkTweenTextureComponent, sdkTweenComponent);
return;
```
And outside the if keep the  `World.Add(entity, sdkTweenComponent);`

in theory should be more efficient as you change the archetype once altogether instead of twice... although, this might get optimized inside the entrails of Arch and make only one change... but still might be easier to understand which components are added to the entity in one case or the other?